Sources/Public/CalendarViewRepresentable.swift

```diff
@@ -0,0 +1,331 @@
+// Created by Bryan Keller on 2/1/23.
+// Copyright © 2023 Airbnb Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import UIKit
+
+// MARK: - CalendarViewRepresentable
+
+/// A declarative, performant calendar `View` that supports use cases ranging from simple date pickers all the way up to
+/// fully-featured calendar apps. Its declarative API makes updating the calendar straightforward, while also providing many
+/// customization points to support a diverse set of designs and use cases.
+///
+/// `CalendarView` does not handle any business logic related to day range selection or deselection. Instead, it provides a
+/// single callback for day selection, allowing you to customize selection behavior in any way that you’d like.
+///
+/// Your business logic can respond to the day selection callback: update any backing-models for your feature. If those backing models
+/// are inputs to your view (for example, `@State`), then the calendar will re-render with the latest data.
+///
+/// - Note: This `View` wraps a UIKit `CalendarView`, enabling it to be used in a SwiftUI hierarchy.
+@available(iOS 13.0, *)
+public struct CalendarViewRepresentable: UIViewRepresentable {
+
+  // MARK: Lifecycle
+
+  /// Initializes a `CalendarViewRepresentable` with default `CalendarItemModel` providers.
+  ///
+  /// - Parameters:
+  ///   - calendar: The calendar on which all date operations will be performed. Defaults to `Calendar.current` (the system's
+  ///   current calendar).
+  ///   - visibleDateRange: The date range that will be displayed. The dates in this range are interpreted using the provided
+  ///   `calendar`.
+  ///   - monthsLayout: The layout of months - either vertically or horizontally.
+  ///   - dataDependency: Any data or state that, when updated, should cause the calendar to re-render. This is needed because
+  ///   `CalendarView` lazily invokes item provider closures outside of the normal SwiftUI update loop. To ensure that the calendar
+  ///   is updated at the right times, pass in any properties that are accessed in your item provider closures. For example, if a
+  ///   `dayItemProvider` closure accesses a local state variable called `selectedDay`, then pass it in here.
+  public init(
+    calendar: Calendar = Calendar.current,
+    visibleDateRange: ClosedRange<Date>,
+    monthsLayout: MonthsLayout,
+    dataDependency: Any? = nil)
```

Comment:
Ah, the `dataDependency`... I thought I could avoid it if I just forced layout, which would trigger our item provider closures to get called, but then I realized the problem is unsolvable without eagerly calling every item provider closure with every possible input. Let me try to explain:

SwiftUI calls the `updateUIView` function, and checks to see which state is referenced during that update pass (imagine every `@State` property has a monitor to see when its `get { ... }` is invoked). SwiftUI then creates a list of dependencies for your view, and any time one of those dependencies changes, your view's `updateUIView` will be invoked again.

For `CalendarView`, you might read state inside the `dayItemProvider` closure, for example, to see if the `@State var selectedDay: Day?` is equal to the current day passed into the closure - and if it is, you change the style of the day to appear selected.

But what if you don't always read the `selectedDay` state in your `dayItemProvider` closure? Maybe your `dayItemProvider` has some code that looks like this:

```swift
.dayItemProvider { day in 
  if day.isCurrentYear {
    return DefaultDayView(day).calendarItemModel
  } else {
    let isSelected = day == selectedDay // The `selectedDay` state is only read here!
    return CustomDayView(day, isSelected: isSelected).calendarItemModel
  }
}
```

When the `CalendarView` is first displayed, it's only going to invoke the `dayItemProvider` for the currently-visible days (likely just 1 or 2 months worth of days). This means that the first time `updateUIView` is called, SwiftUI is going to see that `selectedDay` was never read and is therefore _not_ a dependency for our view. Later on when `selectedDay` changes, `updateUIView` will not be called because `selectedDay` was already determined to not be a dependency.

I don't see a way around this. Maybe there's some kind of hack where we can trigger view updates manually? We still have time to figure out an alternative before I do an official v2.0.0 tag.

Reply:
Hmm, this is a tricky one. At the highest level it seems like the problem is trying to bridge a lazy UIKit-based view to SwiftUI. 

I think the current solution is okay though given the constraints and no other viable options at the moment. Documentation is clear. What if we removed the default value for `dataDependency` to encourage engineers to think about this and avoid potential bugs?

Reply:
> At the highest level it seems like the problem is trying to bridge a lazy UIKit-based view to SwiftUI.
Yep

Good call on removing the default, I agree